src/libraries/System.Runtime/tests/System.Runtime.Tests/System/HalfTests.cs

```diff
@@ -410,6 +410,18 @@ public static IEnumerable<object[]> ExplicitConversion_ToSingle_TestData()
         [Theory]
         public static void ExplicitConversion_ToSingle(Half value, float expected) // Check the underlying bits for verifying NaNs
         {
+            if (PlatformDetection.IsRiscV64Process)
+            {
+                if (BitConverter.HalfToUInt16Bits(value) == 0b0_11111_1010101010)  // Positive Signalling NaN - RISC-V does not preserve payload
+                {
+                    expected = BitConverter.Int32BitsToSingle(0x7FC00000);
+                }
```

Comment:
If different expected values is the right approach, I think it's better to put the condition in ExplicitConversion_ToSingle_TestData.

Reply:
Done